docs/source/extension/extension_dev.rst

```diff
@@ -0,0 +1,651 @@
+- ``page_config`` and ``overrides`` are also handled with traits so that admins can provide defaults and users can provide overrides
+
+
+
+
+Development workflow
+--------------------
+
+We encourage extension authors to add the `jupyterlab-extension GitHub topic
+<https://github.com/search?utf8=%E2%9C%93&q=topic%3Ajupyterlab-extension&type=Repositories>`__ to any GitHub extension repository.
+
+While authoring the extension, you can use the command:
+
+.. code-block:: bash
+
+    npm install   # install npm package dependencies
+    npm run build  # optional build step if using TypeScript, babel, etc.
+    jupyter labextension install  # install the current directory as an extension
+
+This causes the builder to re-install the source folder before building
+the application files. You can re-build at any time using
+``jupyter lab build`` and it will reinstall these packages.
+
+You can also link other local ``npm`` packages that you are working on
+simultaneously using ``jupyter labextension link``; they will be re-installed
+but not considered as extensions. Local extensions and linked packages are
+included in ``jupyter labextension list``.
+
+When using local extensions and linked packages, you can run the command
+
+::
+
+    jupyter lab --watch
+
+This will cause the application to incrementally rebuild when one of the
+linked packages changes. Note that only compiled JavaScript files (and
+the CSS files) are watched by the WebPack process. This means that if
+your extension is in TypeScript you'll have to run a ``jlpm run build``
+before the changes will be reflected in JupyterLab. To avoid this step
+you can also watch the TypeScript sources in your extension which is
+usually assigned to the ``tsc -w`` shortcut. If WebPack doesn't seem to
+detect the changes, this can be related to `the number of available watches <https://github.com/webpack/docs/wiki/troubleshooting#not-enough-watchers>`__.
+
+Note that the application is built against **released** versions of the
+core JupyterLab extensions. If your extension depends on JupyterLab
+packages, it should be compatible with the dependencies in the
+``jupyterlab/static/package.json`` file.  Note that building will always use the latest JavaScript packages that meet the dependency requirements of JupyterLab itself and any installed extensions.  If you wish to test against a
+specific patch release of one of the core JupyterLab packages you can
+temporarily pin that requirement to a specific version in your own
+dependencies.
+
+If you must install an extension into a development branch of JupyterLab, you have to graft it into the source tree of JupyterLab itself. This may be done using the command
+
+::
+
+    jlpm run add:sibling <path-or-url>
```

Comment:
Wondering whether we should also mention the `--extensions-in-dev-mode` flag to load prebuilt extensions.

Reply:
Likely - I haven't started editing this chunk of content yet.

Reply:
Thanks!